src/main/java/net/sourceforge/plantuml/servlet/DiagramResponse.java

```diff
@@ -156,6 +185,23 @@ public void sendDiagram(String uml, int idx) throws IOException {
         diagram.exportDiagram(response.getOutputStream(), idx, new FileFormatOption(format));
     }
 
+    /**
+     * Get PlantUML preprocessor defines.
+     *
+     * @return preprocessor defines
+     */
+    private Defines getPreProcDefines() {
+        final Defines defines;
+        if (SecurityUtils.getSecurityProfile() == SecurityProfile.UNSECURE) {
+            // set dirpath to current dir but keep filename and filenameNoExtension undefined
+            defines = Defines.createWithFileName(new java.io.File("dummy.puml"));
+            defines.overrideFilename("");
+        } else {
+            defines = Defines.createEmpty();
+        }
```

Comment:
Also this part should may be discussed.
As the second part of issue #232 it is noted that the preprocessor function `%dirpath()` never returns a value.